spyder/plugins/editor/widgets/editor.py

```diff
@@ -857,36 +861,42 @@ def clone_from(self, other):
         self.set_stack_index(other.get_stack_index())
 
     @Slot()
-    def open_fileswitcher_dlg(self):
+    @Slot(str)
+    def open_switcher_dlg(self, initial_text=''):
         """Open file list management dialog box"""
         if not self.tabs.count():
             return
-        if self.fileswitcher_dlg is not None and \
-          self.fileswitcher_dlg.is_visible:
-            self.fileswitcher_dlg.hide()
-            self.fileswitcher_dlg.is_visible = False
+        if self.switcher_dlg is not None and self.switcher_dlg.isVisible():
+            self.switcher_dlg.hide()
+            self.switcher_dlg.clear()
             return
-        self.fileswitcher_dlg = FileSwitcher(self, self, self.tabs, self.data,
-                                             ima.icon('TextFileIcon'))
-        self.fileswitcher_dlg.sig_goto_file.connect(self.set_stack_index)
-        self.fileswitcher_dlg.show()
-        self.fileswitcher_dlg.is_visible = True
+        if self.switcher_dlg is None:
+            from spyder.widgets.switcher import Switcher
+            self.switcher_dlg = Switcher(self)
+            self.switcher_manager = EditorSwitcherManager(
+                self.switcher_dlg,
+                lambda: self.get_current_editor(),
+                lambda: self,
+                section=self.get_plugin_title())
+
+        self.switcher_dlg.set_search_text(initial_text)
+        self.switcher_dlg.setup()
+        self.switcher_dlg.show()
+        # Note: the +1 pixel on the top makes it look better
+        delta_top = (self.tabs.tabBar().geometry().height() +
+                     self.fname_label.geometry().height() + 1)
+        self.switcher_dlg.set_position(delta_top)
 
     @Slot()
     def open_symbolfinder_dlg(self):

```

Comment:
The same goes for this method: we still need it? Or can we remove it?

Reply:
Same as above